### DISCUSSION TITLE
PETSc - Architecture Error

### DISCUSSION THREAD

**Discussion by cv319** [firedrakeproject/firedrake#2591] [Firedrake support] [answered]
Hello, I'm having some issues with my firedrake installation. Each time I run firedrake-update, I get to the Installation of PyOP2 I'm greeted by the following error message.
```
/Library/Developer/CommandLineTools/usr/lib/clang/14.0.0/include/mmintrin.h:14:2: error: "This header is only meant to be used on x86 and x64 architecture"

/Library/Developer/CommandLineTools/usr/lib/clang/14.0.0/include/mmintrin.h:314:12: error: invalid conversion between vector type '__m64' (vector of 1 'long long' value) and integer type 'int' of different size
        return (__m64)__builtin_ia32_punpcklwd((__v4hi)__m1, (__v4hi)__m2);
```
As Dr Ham said:
"
That is an x86 only header file which is being pulled in on line 1507 of petscsys.h However the surrounding code is:
```
#if defined(PETSC_HAVE_XMMINTRIN_H) && !defined(__CUDACC__)
#include <xmmintrin.h>
#endif
```
"

I have tried:
- Re-installing command line tools & checking clang was for my machine (pre-M1 Macbook)
- Manually install Py0P2 (but get same errors)
- Tried compiling a simple C program via mpicc (works as expected)
Does anyone know how to handle these flags? :)
[configure.log](https://github.com/firedrakeproject/firedrake/files/9818827/configure.log)
[make.log](https://github.com/firedrakeproject/firedrake/files/9818830/make.log)
[firedrake-update.log](https://github.com/firedrakeproject/firedrake/files/9818835/firedrake-update.log)

**Comment by connorjward** [comment, marked as answer]
Just to check. Are you using a Python installed by homebrew? https://firedrakeproject.org/download.html#system-requirements

**Reply by cv319** [reply]
> Just to check. Are you using a Python installed by homebrew? https://firedrakeproject.org/download.html#system-requirements

yes 😇

**Reply by connorjward** [reply]
The reason I ask is because I'm seeing warnings like:
```
Installing collected packages: PyOP2
  Running setup.py develop for PyOP2
    Running command python setup.py develop
    /Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.9/lib/python3.9/distutils/dist.py:274: UserWarning: Unknown distribution option: 'test_requires'
      warnings.warn(msg)
    running develop
```
This path suggests to me that you're using the system Python. I would have expected the prefix here to be `/opt/homebrew/...` or similar.

**Reply by cv319** [reply]
I managed to make my venv using my system python as supposed to the homebrew version 🤦‍♂️ . Will address this and report back. :)

**Reply by cv319** [reply]
Success! Thank you so much @connorjward  - I really thought I was using homebrew python 😅
